momento-sdk/src/main/java/momento/sdk/CacheClientBuilder.java

```diff
@@ -44,7 +48,23 @@ public CacheClientBuilder setDeadline(@Nonnull Duration deadline) {
         configuration.getTransportStrategy().getGrpcConfiguration().withDeadline(deadline);
     final TransportStrategy newTransportStrategy =
         configuration.getTransportStrategy().withGrpcConfiguration(newGrpcConfiguration);
-    configuration = configuration.withTransportStrategy(newTransportStrategy);
```

Comment:
I understand why we chose to give a ```setDeadline``` option to clients to make them decoupled with our gRPC logic. However, this pattern overrides the configuration object, and needed the below logic to handle not overriding other strategies the client provides. This has a weird anti-pattern effect

Reply:
It seemed an anti pattern to me because the Builder gives an option to setDeadline, which inadvertently overrides the configuration with a new transport strategy. Therefore, if we add more strategies or other things to the configuration, we always need to make sure we don't override the client provided settings in this method. This is already consumed internally, and maybe consumed externally, so i didn't refactor this part. Ideally, the configuration should not be overriden and just set the deadline for the existing strategy.

Reply:
I agree with all the three and 100% agreed on violating the builder pattern. I was skeptical of removing this part to not break backward compatibility, but if we are young then we might take the bullet and do it as this was cause problems in the future as well.